### PR TITLE
get functionality of products

### PR DESCRIPTION
**Pull request by aman-successive**
ticket: https://github.com/nodexpertsdev/RnD/issues/105

### CODE REVIEW

File: expressEngine/product/service/product.js [hunk 1]
```diff
@@ -25,6 +25,27 @@ class Service {
     }
     return { data: product, message: success.productRegistered };
   }
+
+  getProduct = async ({ query, body }) => {
+    const dataToFind = {
+      collection: Product,
+      data: {
+        ...body,
+        isDiscontinued: undefined,
```
Comment (mayank-nx) [inline expressEngine/product/service/product.js]: @aman-successive Either don't set it or set it as false/true depending on the condition

File: expressEngine/cms/product/error.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export default {
   alreadyRegistered: 'You have already registered, Please sign in using provided email id and password',
+  noRecords: 'no records found',
```
Comment (mayank-nx) [inline expressEngine/cms/product/error.js]: @aman-successive Use singular key.

Comment (aman-successive) [reply expressEngine/cms/product/error.js]: Updated it

File: expressEngine/product/service/product.js (hunk 1 above)
Comment (aman-successive) [reply expressEngine/product/service/product.js]: changed it

File: expressEngine/product/service/product.js [hunk 3]
```diff
@@ -25,6 +25,26 @@ class Service {
     }
     return { data: product, message: success.productRegistered };
   }
+
+  getProduct = async ({ query }) => {
+    const dataToFind = {
+      collection: Product,
+      data: {
+        isDiscontinued: { $exists: false },
```
Comment (mayank-nx) [inline expressEngine/product/service/product.js]: @aman-successive it is still wrong. Set check as isDiscontinued is not true

Comment (aman-successive) [reply expressEngine/product/service/product.js]: updated it now

File: expressEngine/cms/product/error.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export default {
   alreadyRegistered: 'You have already registered, Please sign in using provided email id and password',
+  noRecord: 'no record found',
```
Comment (chiragSuccessive) [inline expressEngine/cms/product/error.js]: @aman-successive Please place noRecord in common cms.

Comment (aman-successive) [reply expressEngine/cms/product/error.js]: updated it

File: expressEngine/product/service/product.js [hunk 5]
```diff
@@ -25,6 +26,26 @@ class Service {
...
+    const dataToFind = {
+      collection: Product,
+      data: {
+        isDiscontinued: { $ne: true },
+      },
+      limit: query.limit,
+      skip: query.skip,
+    };
+    const products = (await DBService.find(dataToFind));
+    const err = { error: error.noRecords };
```
Comment (chiragSuccessive) [inline expressEngine/product/service/product.js]: @aman-successive Please place err object in if condition using it.

Comment (aman-successive) [reply expressEngine/product/service/product.js]: updated it

**Pull request merged**